File: android/app/src/main/kotlin/com/aws/guardian/models/Cost.kt
package com.aws.guardian.models

import com.google.firebase.database.IgnoreExtraProperties
import java.io.Serializable

@IgnoreExtraProperties
data class Cost(
    val id: String = "",
    val date: Long = 0L,
    val amount: Double = 0.0,
    val service: String = "",
    val account: String = ""
) : Serializable

data class CostSummary(
    val dailyCost: Double = 0.0,
    val monthlyCost: Double = 0.0,
    val trend: Double = 0.0,  // percentage change
    val byService: Map<String, Double> = emptyMap(),
    val byAccount: Map<String, Double> = emptyMap()
)

data class CostHistory(
    val dates: List<String> = emptyList(),
    val amounts: List<Double> = emptyList()
)

File: ios/GuardianApp/Models/Cost.swift
import Foundation

struct Cost: Codable, Identifiable {
    let id: String
    let date: Date
    let amount: Double
    let service: String
    let account: String

    enum CodingKeys: String, CodingKey {
        case id
        case date
        case amount
        case service
        case account
    }
}

struct CostSummary: Codable {
    let dailyCost: Double
    let monthlyCost: Double
    let trend: Double // percentage change
    let byService: [String: Double]
    let byAccount: [String: Double]
}

struct CostHistory: Codable {
    let dates: [String]
    let amounts: [Double]
}
